### ISSUE TITLE
Use `--depth 1` when cloning git repos with no `ref` specified

### ISSUE THREAD

**Issue by psss** [teemtee/tmt#1237]
In order to speed up testing and minimize data transfers let's use `git clone --depth 1` when cloning git repositories and no specific `ref` is requested by the user.

**Comment by idorax** [comment]
There are two lines involved in `git clone`, e.g.
```bash
$ find . -type f | xargs grep 'clone'
./beakerlib.py:    dest ........ target folder into which the library repo is cloned
./beakerlib.py:            # Prepare path, clone the repository, checkout ref
./beakerlib.py:                        ['git', 'clone', self.url, directory],
./steps/discover/fmf.py:                ['git', 'clone', url, self.testdir],
./utils.py:            ('git clone', f"repo '{fmf_id.get('url')}' cannot be cloned"),
```

@psss, we need to update these two lines when running `git clone`, right?
1.  [steps/discover/fmf.py](https://github.com/idorax/tmt/blob/main/tmt/steps/discover/fmf.py#L274)
    ```python
            self.run(
                ['git', 'clone', url, self.testdir],
                env={"GIT_ASKPASS": "echo"})
    ```
1. [tmt/beakerlib.py](https://github.com/idorax/tmt/blob/main/tmt/beakerlib.py#L166)
    ```python
                if self.url:
                    self.parent.run(
                        ['git', 'clone', self.url, directory],
                        env={"GIT_ASKPASS": "echo"})
    ```

**Comment by psss** [comment]
Yes, both these locations should be adjusted with the `--depth 1` option.

**Comment by idorax** [comment]
To fix [steps/discover/fmf.py](https://github.com/idorax/tmt/blob/main/tmt/steps/discover/fmf.py#L274), we should organize different commands according to the value of `ref`. If `ref` is `None`, the commands should include `--depth 1`. e.g.
```python
            if ref is None:
                commands = ['git', 'clone', url, '--depth 1', self.testdir]
            else:
                commands = ['git', 'clone', url, self.testdir]
            self.run(commands, env={"GIT_ASKPASS": "echo"})
```

Similarly，[tmt/beakerlib.py](https://github.com/idorax/tmt/blob/main/tmt/beakerlib.py#L166) by be updated to speed up testing.

**Comment by idorax** [comment]
@psss and @lukaszachy, the proposed fix is ready, please help to review PR#1253, thanks!
* https://github.com/teemtee/tmt/pull/1253

**Comment by lukaszachy** [comment]
Thank you for taking care of this issue @idorax .
